pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -0,0 +1,474 @@
+
+    credman_credential_t credential;
+    ret = credman_get(&credential, sock->tag, CREDMAN_TYPE_PSK);
+    if (ret < 0) {
+        DEBUG("sock_dtls: no matching PSK credential found\n");
+        return dtls_alert_fatal_create(DTLS_ALERT_DECRYPT_ERROR);
+    }
+
+    const void *c = NULL;
+    size_t c_len = 0;
+    switch(type) {
+    case DTLS_PSK_HINT:
+        DEBUG("sock_dtls: psk hint request\n");
+        /*
+        c = credential.params.psk.hint.s;
+        c_len = credential.params.psk.hint.len;
+        */
+        /* server: unused */
+        return 0;
+    case DTLS_PSK_IDENTITY:
+        DEBUG("sock_dtls: psk id request\n");
+        c = credential.params.psk.id.s;
+        c_len = credential.params.psk.id.len;
+        break;
+    case DTLS_PSK_KEY:
+        DEBUG("sock_dtls: psk key request\n");
+        c = credential.params.psk.key.s;
+        c_len = credential.params.psk.key.len;
+        break;
+    default:
+        DEBUG("sock:dtls unsupported request type: %d\n", type);
+        return dtls_alert_fatal_create(DTLS_ALERT_INTERNAL_ERROR);
+    }
+    if (c_len > result_length) {
+        DEBUG("sock_dtls: not enough memory for credential type: %d\n", type);
+        return dtls_alert_fatal_create(DTLS_ALERT_INTERNAL_ERROR);
+    }
+    if (c == NULL || c_len == 0) {
+        DEBUG("sock_dtls: invalid credential params for type %d\n", type);
+        return dtls_alert_fatal_create(DTLS_ALERT_INTERNAL_ERROR);
+    }
+    memcpy(result, c, c_len);
+    return c_len;
+}
+#endif /* DTLS_PSK */
+
+#ifdef DTLS_ECC
+static int _get_ecdsa_key(struct dtls_context_t *ctx, const session_t *session,
+                          const dtls_ecdsa_key_t **result)
+{
+    (void)session;
+    int ret;
+    sock_dtls_t *sock = (sock_dtls_t *)dtls_get_app_data(ctx);
+
+    credman_credential_t credential;
+    ret = credman_get(&credential, sock->tag, CREDMAN_TYPE_ECDSA);
+    if (ret < 0) {
+            DEBUG("sock_dtls: no matching ecdsa credential found\n");
+            return dtls_alert_fatal_create(DTLS_ALERT_INTERNAL_ERROR);
+    }
+
+    static dtls_ecdsa_key_t key;
+    key.curve = DTLS_ECDH_CURVE_SECP256R1;
+    key.priv_key = credential.params.ecdsa.private_key;
+    key.pub_key_x = credential.params.ecdsa.public_key.x;
+    key.pub_key_y = credential.params.ecdsa.public_key.y;
+    *result = &key;
+    return 0;
+}
+
+static int _verify_ecdsa_key(struct dtls_context_t *ctx,
+                             const session_t *session,
+                             const unsigned char *other_pub_x,
+                             const unsigned char *other_pub_y, size_t key_size)
+{
+    (void) ctx;
+    (void) session;
+    (void) other_pub_y;
+    (void) other_pub_x;
+    (void) key_size;
+
+    return 0;
+}
+#endif /* DTLS_ECC */
+
+int sock_dtls_create(sock_dtls_t *sock, sock_udp_t *udp_sock,
+                     credman_tag_t tag, unsigned version, unsigned role)
+{
+    assert(sock && udp_sock);
```

Comment:
please use single conditions for `assert`, applies below too

Reply:
fixed.